src/cascade/model/operations.py

```diff
@@ -21,68 +21,33 @@ def _var_grid_iterator(var_df):
-        to be constant are in the var table.
-
     Args:
         var_df (pd.DataFrame): The DB.var table from the Dismod-AT database.
 
     Returns:
         (str, (int, int), pd.DataFrame): the kind of grid, its unique
         identifier, and the grid itself.
     """
-    for smooth_id, sub_grid_df in var_df.groupby("smooth_id"):
-        rate_grids = sub_grid_df[
-            (sub_grid_df.var_type == "rate") & (sub_grid_df.rate_id >= 0)][
-            ["node_id", "rate_id"]].drop_duplicates()
-        meas_value_full = sub_grid_df[(sub_grid_df.var_type == "mulcov_meas_value") & (sub_grid_df.covariate_id >= 0)]
-        meas_value = meas_value_full[["integrand_id", "covariate_id"]].drop_duplicates()
-        meas_std_full = sub_grid_df[(sub_grid_df.var_type == "mulcov_meas_std") & (sub_grid_df.covariate_id >= 0)]
-        meas_std = meas_std_full[["integrand_id", "covariate_id"]].drop_duplicates()
-        rate_value_full = sub_grid_df[(sub_grid_df.var_type == "mulcov_rate_value") & (sub_grid_df.covariate_id >= 0)]
-        rate_value = rate_value_full[["rate_id", "covariate_id"]].drop_duplicates()
-
-        categories = [rate_grids, meas_value, meas_std, rate_value]
-        category_cnt = sum([not test_kind.empty for test_kind in categories])
-        if category_cnt > 1:
-            raise RuntimeError(
-                f"The var table for smooth_id={smooth_id} should have only one kind "
-                f"but seems to have multiple kinds: rate? {not rate_grids.empty} meas value {not meas_value.empty} "
-                f"meas std? {not meas_std.empty} rate_value {not rate_value.empty}"
-            )
-        elif category_cnt == 0:
-            raise RuntimeError(
-                f"Cannot figure out why smooth_id={smooth_id} isn't one of "
-                f"a rate or a covariate multiplier")
-        # else: It's OK
-
-        if not rate_grids.empty:
-            node_id, rate_id = [int(rid) for rid in rate_grids.iloc[0]]
-            yield "rate", (smooth_id, rate_id, node_id), sub_grid_df
-
-        elif not meas_value.empty:
-            integrand_id, covariate_id = [int(meas_id) for meas_id in meas_value.iloc[0]]
-            yield "meas_value", (smooth_id, integrand_id, covariate_id), sub_grid_df
-
-        elif not meas_std.empty:
-            integrand_id, covariate_id = [int(std_id) for std_id in
-                                          meas_std.iloc[0]]
-            yield "meas_std", (smooth_id, integrand_id, covariate_id), sub_grid_df
-
-        elif not rate_value.empty:
-            rate_id, covariate_id = [int(rval_id) for rval_id in
-                                     rate_value.iloc[0]]
-            yield "rate_value", (smooth_id, rate_id, covariate_id), sub_grid_df
-
+    # groupby ignores every record where any of the keys are Null, so it
+    # only gets the rates.
+    for rate_index, sub_grid_df in var_df.groupby(["smooth_id", "rate_id", "node_id"]):
```

Comment:
Does this get rates and rate_value?

Reply:
It doesn't get rate value because the node_id is null for rate value, so it doesn't show up in the gruop by.